tests/response/test_streaming_response.py

```diff
@@ -0,0 +1,142 @@
+"""A large part of the tests in this file were adapted from:
+
+https://github.com/encode/starlette/blob/master/tests/test_responses.py And are
+meant to ensure our compatibility with their API.
+"""
+from typing import TYPE_CHECKING, AsyncIterator, Iterator
+
+import anyio
+
+from starlite import BackgroundTask
+from starlite.response import StreamingResponse
+from starlite.testing import TestClient
+
+if TYPE_CHECKING:
+    from starlite.types import Message, Receive, Scope, Send
+
+
+def test_streaming_response_unknown_size() -> None:
+    app = StreamingResponse(content=iter(["hello", "world"]))
+    client = TestClient(app)
+    response = client.get("/")
+    assert "content-length" not in response.headers
+
+
+def test_streaming_response_known_size() -> None:
+    app = StreamingResponse(content=iter(["hello", "world"]), headers={"content-length": "10"})
+    client = TestClient(app)
+    response = client.get("/")
+    assert response.headers["content-length"] == "10"
+
+
+async def test_streaming_response_stops_if_receiving_http_disconnect(anyio_backend: str) -> None:
+    streamed = 0
+
+    disconnected = anyio.Event()
+
+    async def receive_disconnect() -> dict:
+        await disconnected.wait()
+        return {"type": "http.disconnect"}
+
+    async def send(message: "Message") -> None:
+        nonlocal streamed
+        if message["type"] == "http.response.body":
+            streamed += len(message.get("body", b""))
+            # Simulate disconnection after download has started
+            if streamed >= 16:
+                await disconnected.set()
+
+    async def stream_indefinitely() -> AsyncIterator[bytes]:
+        while True:
+            # Need a sleep for the event loop to switch to another task
+            await anyio.sleep(0)
+            yield b"chunk "
+
+    response = StreamingResponse(content=stream_indefinitely())
+
+    with anyio.move_on_after(1) as cancel_scope:
+        await response({}, receive_disconnect, send)  # type: ignore
+    assert not cancel_scope.cancel_called, "Content streaming should stop itself."
+
+
+def test_streaming_response() -> None:
```

Comment:
This should be tested with a synchronous generator as well 
(also applies to the following tests)

Reply:
there are tests with sync generators elsewhere that use this, but i will add

Reply:
Must have missed them! Maybe move them here then because that's where you'd expect them (at least I did)